nflog/nflog.go

```diff
@@ -289,48 +216,79 @@ func marshalMeshEntry(e *pb.MeshEntry) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+// Options configures a new Log implementation.
+type Options struct {
+	SnapshotReader io.Reader
+	SnapshotFile   string
+
+	Retention time.Duration
+
+	Logger  log.Logger
+	Metrics prometheus.Registerer
+}
+
+func (o *Options) validate() error {
+	if o.SnapshotFile != "" && o.SnapshotReader != nil {
+		return errors.New("only one of SnapshotFile and SnapshotReader must be set")
+	}
+
+	return nil
+}
+
 // New creates a new notification log based on the provided options.
 // The snapshot is loaded into the Log if it is set.
-func New(opts ...Option) (*Log, error) {
+func New(o Options) (*Log, error) {
+	if err := o.validate(); err != nil {
+		return nil, err
+	}
+
+	if o.SnapshotFile != "" {
+		if r, err := os.Open(o.SnapshotFile); err != nil {
+			if !os.IsNotExist(err) {
```

Comment:
Should we maybe log at info-level that a previous log to load up didn't exist, so it'll create one? Or, maybe the inverse path, something like "Loading a previous snapshot..."

At a minimum, we should log it at debug level, since there are sort of two independent paths here. If for some reason a file can't be accessed, things could end up in a weird state, and there is no evidence trail left behind that it happened.